### DISCUSSION TITLE
Correct syntax for switching between inline adapters?

### DISCUSSION THREAD

**Discussion by r-terrett** [olimorris/codecompanion.nvim#516] [Q&A] [answered]
Hi,

I'm trying to switch between a local ollama adapter and an anthropic adapter, defined thusly:

```lua
		adapters = {
			ollama = function()
				return require("codecompanion.adapters").extend("ollama", {
					name = "qwen",
					env = {
						url = [REDACTED],
					},
					headers = {
						["Content-Type"] = "application/json",
						["Authorization"] = "Bearer ${api_key}",
					},
					parameters = {
						sync = true,
					},
					schema = {
						model = {
							default = "qwen2.5-coder:32b",
						},
					},
				})
			end,
			anthropic = function()
				return require("codecompanion.adapters").extend("anthropic", {
					name = "claude",
					schema = {
						model = {
							default = "claude-3-5-sonnet-20241022",
						},
						max_tokens = { default = 4096 },
					},
				})
			end,
		},
```

These adapters work fine in isolation, and with default strategies set to 'ollama', inline editing is handled by Qwen2.5-coder. I am able to use `ga` to switch adapters in a chat pane, however this does not change the inline editing adapter.

The README provides advice that the adapter may be selected by setting the `vim.g.codecompanion_adapter` variable. I find that this variable is `nil` unless I set it. Upon setting it using (for example) `:let vim.g.codecompanion_adapter="anthropic"` I find that CodeCompanion fails on attempted inline use, with:

`Error executing Lua callback: ...decompanion.nvim/lua/codecompanion/strategies/inline.lua:135: attempt to index field 'adapter' (a function value)`

It's unclear how to select an adapter using the `codecompanion_adapter` variable. I would appreciate any guidance which can be provided here and would suggest that the documentation may bear additional explanation of this feature (however I don't know much about Lua, so maybe what I'm doing wrong is obvious).

Thanks in advance.

**Comment by olimorris** [comment, marked as answer]
Moved this to #518
